crates/pathfinder/src/bin/pathfinder/config.rs

```diff
@@ -303,7 +304,13 @@ This should only be enabled for debugging purposes as it adds substantial proces
 
     #[arg(
         long = "rpc.custom-versioned-constants-json-path",
-        long_help = "Path to a JSON file containing the versioned constants to use for execution",
+        long_help = "Path to a JSON file referencing sequencer versioned constants. The file maps \
+                     Starknet version keys to paths of files containing the versioned constants, \
+                     which are then used for blocks with that version _and_ subsequent versions \
+                     smaller than a hard-coded versioned constants change (or the next key in the \
+                     custom map). Alternatively, for backwards compatibility, path to the version \
+                     constants file can be passed directly, in which case it's used for the \
+                     latest version.",
```

Comment:
"smaller than a hard-coded versioned constants change" -- I don't quite get this part. What do you mean here?

I think we should also mentioned that the paths to the individual versioned constants JSON files should be _relative_ to the path of this file.

Reply:
well, you can have custom config like
```
{
    "0.13.3": "my_versioned_constants_0_13_3.json"
}
```
and it overrides the versioned constants for `0.13.3`, but not for `0.13.4` - presumably Starkware changed them because they were missing something, and if you took the base for `my_versioned_constants_0_13_3.json` from Starkware, using it for `0.13.4` might break. So, as implemented, the lookup reverts to using defaults at `0.13.4` (the "hard-coded change"), and if you want your custom config to not revert, you have to create a new one and add it: 
```
{
    "0.13.3": "my_versioned_constants_0_13_3.json",
    "0.13.4": "my_versioned_constants_0_13_4.json",
}
```

Reply:
So effectively whatever you configure in the JSON are just overrides for specific versions, right? They're effectively merged with the defaults, overriding specific versions.

Reply:
yes, exactly

Reply:
plus, there is flexibility in having multiple versions where Starknet doesn't - i.e. you can have
```
{
    "0.13.3.1": "my_versioned_constants_0_13_3_1.json",
    "0.13.3.2": "my_versioned_constants_0_13_3_2.json"
}
```
not sure if anybody needs that, but IMHO it's simpler to allow it than to document why it's forbidden... :-)

Reply:
> I think we should also mentioned that the paths to the individual versioned constants JSON files should be _relative_ to the path of this file.

They _can_ be relative, but yes...